commit 55607e1e5c2cc3dba721062307cc5a346a15e915
Author: Mart Raudsepp <leio@gentoo.org>
Date:   Sun Mar 15 23:53:33 2020 +0100

    build: Make printers panel (cups) optional
    
    As a side-effect smbclient is also optional, as it's only required
    by the printers panel.

diff --git a/meson.build b/meson.build
index d88c10354..63c2a4859 100644
--- a/meson.build
+++ b/meson.build
@@ -161,9 +161,10 @@ common_deps = [
 ]
 
 # Check for CUPS 1.4 or newer
-cups_dep = dependency('cups', version : '>= 1.4', required: false)
-assert(cups_dep.found(), 'CUPS 1.4 or newer not found')
+cups_dep = dependency('cups', version : '>= 1.4', required: get_option('cups'))
+enable_cups = cups_dep.found()
 
+if enable_cups
 # https://bugzilla.gnome.org/show_bug.cgi?id=696766
 cups_cflags = []
 if cups_dep.version().version_compare('>= 1.6')
@@ -179,6 +180,9 @@ check_headers = [
 foreach header: check_headers
   assert(cc.has_header(header[1], args: cups_cflags), 'CUPS headers not found: ' + header[1])
 endforeach
+endif
+config_h.set('BUILD_PRINTERS', enable_cups,
+             description: 'Define to 1 to build the Printers panel')
 
 config_h.set10('HAVE_CUPS_HTTPCONNECT2',
                cc.has_function('httpConnect2', dependencies: cups_dep),
diff --git a/meson_options.txt b/meson_options.txt
index 8f3b73365..00d5fec9a 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,5 +1,6 @@
 option('bluetooth', type: 'boolean', value: true, description: 'build with Bluetooth support')
 option('cheese', type: 'boolean', value: true, description: 'build with cheese webcam support')
+option('cups', type: 'feature', value: 'auto', description: 'build with CUPS support (printer panel)')
 option('documentation', type: 'boolean', value: false, description: 'build documentation')
 option('goa', type: 'feature', value: 'auto', description: 'build with gnome-online-accounts support')
 option('grilo', type: 'feature', value: 'auto', description: 'build with grilo support (background panel)')
diff --git a/panels/meson.build b/panels/meson.build
index 90bb3deef..7d2660986 100644
--- a/panels/meson.build
+++ b/panels/meson.build
@@ -17,7 +17,6 @@ panels = [
   'mouse',
   'notifications',
   'power',
-  'printers',
   'region',
   'removable-media',
   'search',
@@ -28,6 +27,10 @@ panels = [
   'user-accounts'
 ]
 
+if enable_cups
+  panels += ['printers']
+endif
+
 if enable_goa
   panels += ['online-accounts']
 endif
diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index b10611c80..cce44eeec 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -52,7 +52,9 @@ extern GType cc_notifications_panel_get_type (void);
 extern GType cc_goa_panel_get_type (void);
 #endif /* BUILD_GOA */
 extern GType cc_power_panel_get_type (void);
+#ifdef BUILD_PRINTERS
 extern GType cc_printers_panel_get_type (void);
+#endif
 extern GType cc_region_panel_get_type (void);
 extern GType cc_removable_media_panel_get_type (void);
 extern GType cc_search_panel_get_type (void);
@@ -118,7 +120,9 @@ static CcPanelLoaderVtable default_panels[] =
   PANEL_TYPE("online-accounts",  cc_goa_panel_get_type,                  NULL),
 #endif
   PANEL_TYPE("power",            cc_power_panel_get_type,                NULL),
+#ifdef BUILD_PRINTERS
   PANEL_TYPE("printers",         cc_printers_panel_get_type,             NULL),
+#endif
   PANEL_TYPE("region",           cc_region_panel_get_type,               NULL),
   PANEL_TYPE("removable-media",  cc_removable_media_panel_get_type,      NULL),
   PANEL_TYPE("search",           cc_search_panel_get_type,               NULL),
diff --git a/tests/meson.build b/tests/meson.build
index 8410c636d..a727b0caa 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -6,5 +6,7 @@ endif
 
 subdir('interactive-panels')
 
-subdir('printers')
+if enable_cups
+  subdir('printers')
+endif
 subdir('info')
